### DISCUSSION TITLE
Unable to upload build to App store using fastlane

### DISCUSSION THREAD

**Discussion by lavanyaRamesh064** [fastlane/fastlane#21046] [Q&A]
Hi,
I have been getting the below error, when try to upload a build to app store connect. I have reproduced steps to Create new API Keys to Use With the App Store Server API but getting same error

Authentication credentials are missing or invalid.
  - Provide a properly configured and signed bearer token, and make sure that it has not expired.
    Learn more about Generating Tokens for API Requests https://developer.apple.com/go/?id=api-generating-tokens

**Comment by mdgeist** [comment]
I've been experiencing something similar. Curious how are you using your Api Key in Fastlane?

Theres `pilot`, you can also call `upload_to_testflight`. Even with that you can also explicitly call `app_store_connect_api_key` and Fastlane will store that in a SharedValue or you can get the api_key and pass that to pilot and `upload_to_testflight`.

In my case subsequent retries will get it to work eventually... It's really annoying though and it's been happening since late last year.

**Comment by david-var** [comment]
it should be working now. please check this:
https://github.com/fastlane/fastlane/issues/21060#issuecomment-1434916276
